components/dash-table/src/dash-table/utils/TableClipboardHelper.ts

```diff
@@ -111,4 +111,13 @@ export default class TableClipboardHelper {
             overflowRows
         );
     }
+
+    private static parse(str: string) {
+        const temprows = str.split('\n');
+        if (temprows.length > 1 && temprows[temprows.length - 1] === '') {
+            temprows.pop();
+        }
+        const rows = temprows.map(row => row.split('\t'));
```

Comment:
Thanks for tracking down the original [`SheetClip.parse`](https://github.com/warpech/sheetclip/blob/f6fc218347e37ad6e630e01c757a83e2c4055445/sheetclip.js#L21). I'm a little worried that this simple replacement - while fixing the problem you encountered with multiple quotes within a cell - will cause other problems due to extra cases that were covered by `SheetClip` and missed here. I'm thinking particularly about cells that have tab or newline characters in them, which presumably are the whole reason for all that monkey business with quotes, so if a cell starts and ends with quote characters it can have tabs and newlines inside it and these don't cause the parser to move to the next column or row.

Then of course we could ask what happens if there are ALSO quote characters inside the cell text. I'd have guessed these would be escaped (with a backslash before the quote) but I see no code in the original `SheetClip.parse` to handle that so perhaps the quote character is doubled? That would explain their "replace two quotes with one" step, it's just applied too broadly.

Anyway I think the goal of `SheetClip` as expressed in its [head comment](https://github.com/warpech/sheetclip/blob/f6fc218347e37ad6e630e01c757a83e2c4055445/sheetclip.js#L5) is our goal as well:
> This tiny library transforms JavaScript arrays to strings that are pasteable by LibreOffice, OpenOffice, Google Docs and Microsoft Excel.

Meaning we want to be able to copy and paste rectangular sections of dash table to and from common spreadsheet programs, as well as to other dash tables, regardless of what text is in those cells. So I think this requires a little more investigation about how such cells are encoded by spreadsheets - and it may mean we have more work to do on the copy side as well as on paste.

Reply:
That makes a lot of sense -- I was wondering if this solution was too simplistic. 

One thought I have (which may also be too simplistic) is that when parsing pasted cells we could first "escape" quotes by adding a backslash before each quote and then keep the SheetClip.parse function the same except for changing `.replace(/""/g, '"')` steps to something like `.replace('\"', '"')`. Does that general idea sound okay to you or am I oversimplifying?

Regardless, I'm happy to keep looking at this. Thank you again for your help!